components/script/dom/globalscope.rs

```diff
@@ -2227,6 +2231,66 @@ impl GlobalScope {
         }))
     }
 
+    pub fn create_image_bitmap(
+        &self,
+        image: ImageBitmapSource,
+        options: &ImageBitmapOptions,
+    ) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+        if options.resizeWidth.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        if options.resizeHeight.unwrap() == 0 {
+            p.reject_error(Error::InvalidState)
+        }
+
+        let image_bitmap = ImageBitmap::new(&global, 0, 0).unwrap();
+
+        let promise = match image {
+            ImageBitmapSource::HTMLCanvasElement(ref canvas) => {
+                // https://html.spec.whatwg.org/multipage/#check-the-usability-of-the-image-argument
+                if !canvas.is_valid() {
+                    p.reject_error(Error::InvalidState)
```

Comment:
We're going to want to return p at this point, otherwise we'll still fetch the canvas data and update the image bitmap contents.

Reply:
That's clear. Got it.